src/test/java/com/service/OrderServiceTest.java

```diff
@@ -0,0 +1,159 @@
+        verify(dishDAOMock, times(2)).getById(anyLong());
+
+        assertEquals(act, exp);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void showOrdersWithDetailsNoUser() {
+
+        when(userDAOMock.getByName("Ivan")).thenReturn(Optional.ofNullable(null));
+        service.showOrdersWithDetails("Ivan", Order.Status.CREATED);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void showOrdersWithDetailsNoOrders() {
+
+        when(userDAOMock.getByName("Ivan")).thenReturn(Optional.of(users.get(0)));
+        when(orderDAOMock.getAll()).thenReturn(new ArrayList<>());
+
+        service.showOrdersWithDetails("Ivan", Order.Status.CREATED);
+    }
+
+
+    @Test
+    public void cancelOrder() throws Exception {
+        when(orderDAOMock.remove(anyLong())).thenReturn(true);
+
+        boolean check = service.cancelOrder(anyLong());
+        verify(orderDAOMock, times(1)).remove(anyLong());
+
+        assertTrue(check);
+    }
+
+
+    @Test(expected = NoSuchElementException.class)
+    public void makeOrderNoUser() throws Exception {
+
+        when(userDAOMock.getByName("Ivan")).thenReturn(Optional.ofNullable(null));
+        service.makeOrder("Ivan", userChoice);
+    }
+
+    @Test
+    public void makeOrderValid() {
+
+        when(userDAOMock.getByName("Ivan")).thenReturn(Optional.of(users.get(0)));
+        when(orderDAOMock.create(new Order(0, users.get(0).getId(), LocalDateTime.now(), 0, Order.Status.CREATED)))
+                .thenReturn(true);
+
+        when(orderDAOMock.getAll()).thenReturn(orders);
+        when(dishDAOMock.getByName("MOJITO")).thenReturn(Optional.of(dishes.get(0)));
+        when(dishDAOMock.getByName("BURRITO")).thenReturn(Optional.of(dishes.get(1)));
+
+        DishOrder dishOrder1 = new DishOrder(0, orders.get(0).getId(),
+                dishes.get(0).getId(), amount_1, amount_1 * dishes.get(0).getPrice());
+        when(dishOrderDAOMock.create(dishOrder1)).thenReturn(true);
+
+        DishOrder dishOrder2 = new DishOrder(0, orders.get(0).getId(),
+                dishes.get(1).getId(), amount_2, amount_2 * dishes.get(1).getPrice());
+        when(dishOrderDAOMock.create(dishOrder2)).thenReturn(true);
+
+        when(orderDAOMock.getById(orders.get(0).getId())).thenReturn(Optional.of(orders.get(0)));
+        when(orderDAOMock.update(orders.get(0))).thenReturn(true);
+
+        service.makeOrder("Ivan", userChoice);
+
+        verify(userDAOMock, times(2)).getByName("Ivan");
+        verify(orderDAOMock, times(1)).create(any());
+        verify(orderDAOMock, times(1)).getAll();
+        verify(dishDAOMock, times(1)).getByName("MOJITO");
+        verify(dishDAOMock, times(1)).getByName("BURRITO");
+        verify(dishOrderDAOMock, times(1)).create(dishOrder1);
+        verify(dishOrderDAOMock, times(1)).create(dishOrder2);
+        verify(orderDAOMock, times(1)).getById(orders.get(0).getId());
+        verify(orderDAOMock, times(1)).update(orders.get(0));
+
```

Comment:
and one more time

Reply:
Here we use 'times(wantedNumberOfInvocations t)' method. Usage of 'verifyNoMoreInteractions' will be redundant.